test/e2e/jmx/jmx_test.go

```diff
@@ -0,0 +1,364 @@
+		validateMetrics(t, []string{
+			"kafka.message.count",
+			"kafka.request.count",
+			"kafka.request.failed",
+			"kafka.request.time.total",
+			"kafka.request.time.50p",
+			"kafka.request.time.99p",
+			"kafka.request.time.avg",
+			"kafka.consumer.fetch-rate",
+			"kafka.consumer.total.bytes-consumed-rate",
+			"kafka.consumer.total.records-consumed-rate",
+			"kafka.producer.io-wait-time-ns-avg",
+			"kafka.producer.outgoing-byte-rate",
+			"kafka.producer.response-rate",
+		}, "KAFKA_E2E")
+	})
+}
+
+func testContainerInsightsMetrics(t *testing.T) {
+	t.Run("verify_containerinsights_metrics", func(t *testing.T) {
+		validateMetrics(t, []string{
+			"jvm_classes_loaded",
+			"jvm_threads_current",
+			"jvm_threads_daemon",
+			"java_lang_operatingsystem_totalswapspacesize",
+			"java_lang_operatingsystem_systemcpuload",
+			"java_lang_operatingsystem_processcpuload",
+			"java_lang_operatingsystem_freeswapspacesize",
+			"java_lang_operatingsystem_totalphysicalmemorysize",
+			"java_lang_operatingsystem_freephysicalmemorysize",
+			"java_lang_operatingsystem_openfiledescriptorcount",
+			"java_lang_operatingsystem_availableprocessors",
+			"jvm_memory_bytes_used",
+			"jvm_memory_pool_bytes_used",
+			"catalina_manager_activesessions",
+			"catalina_manager_rejectedsessions",
+			"catalina_globalrequestprocessor_requestcount",
+			"catalina_globalrequestprocessor_errorcount",
+			"catalina_globalrequestprocessor_processingtime",
+		}, "ContainerInsights/Prometheus")
+	})
+}
+
+func testTomcatRejectedSessions(t *testing.T) {
+	t.Run("verify_catalina_manager_rejectedsessions", func(t *testing.T) {
+		generateTraffic(t)
+		time.Sleep(wait)
+		verifyMetricAboveZero(t, "catalina_manager_rejectedsessions", "ContainerInsights/Prometheus", true)
+	})
+}
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+func validateMetrics(t *testing.T, metrics []string, namespace string) {
+	for _, metric := range metrics {
+		t.Run(metric, func(t *testing.T) {
+			awsservice.ValidateMetricWithTest(t, metric, namespace, nil, 5, interval)
+		})
+	}
+}
+
+func generateTraffic(t *testing.T) {
+	cmd := exec.Command("kubectl", "get", "nodes", "-o", "jsonpath='{.items[0].status.addresses[?(@.type==\"ExternalIP\")].address}'")
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err, "Error getting node external IP")
+
+	nodeIP := strings.Trim(string(output), "'")
+	require.NotEmpty(t, nodeIP, "Node IP failed to format")
+
+	for i := 0; i < 5; i++ {
+		resp, err := http.Get(fmt.Sprintf("http://%s:30080/webapp/index.jsp", nodeIP))
+		if err != nil {
+			t.Logf("Request attempt %d failed: %v", i+1, err)
+			continue
+		}
+		require.NoError(t, resp.Body.Close(), "Failed to close response body")
+	}
+}
+
+func verifyMetricAboveZero(t *testing.T, metricName, namespace string, containerInsights bool) {
```

Comment:
So these tests are just checking the existence of metrics, not a value.

Reply:
Most of the tests in `jmx_test.go` check for the existence of metrics to ensure the agent properly emits them. However, for `tomcat.sessions` and `tomcat.rejected_sessions`, the `verifyMetricAboveZero` function does more: it verifies if these metrics increase above 0 when we send load to the Tomcat application. This additional check was an important part of manual testing to ensure these specific metrics can emit non-zero values, which wasn't a concern for other metrics.